pkg/quota/image/imagestreamimport_evaluator.go

```diff
@@ -0,0 +1,81 @@
+package image
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kquota "k8s.io/kubernetes/pkg/quota"
+	"k8s.io/kubernetes/pkg/quota/generic"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const imageStreamImportName = "Evaluator.ImageStreamImport"
+
+// NewImageStreamImportEvaluator computes resource usage for ImageStreamImport objects. This particular kind
+// is a virtual resource. It depends on ImageStream usage evaluator to compute image numbers before the
+// the admission can work.
+func NewImageStreamImportEvaluator(isNamespacer osclient.ImageStreamsNamespacer) kquota.Evaluator {
+	computeResources := []kapi.ResourceName{
+		imageapi.ResourceImageStreams,
+	}
+
+	matchesScopeFunc := func(kapi.ResourceQuotaScope, runtime.Object) bool { return true }
+
+	return &generic.GenericEvaluator{
+		Name:                       imageStreamImportName,
+		InternalGroupKind:          imageapi.Kind("ImageStreamImport"),
+		InternalOperationResources: map[admission.Operation][]kapi.ResourceName{admission.Create: computeResources},
+		MatchedResourceNames:       computeResources,
+		MatchesScopeFunc:           matchesScopeFunc,
+		UsageFunc:                  makeImageStreamImportAdmissionUsageFunc(isNamespacer),
+		ListFuncByNamespace: func(namespace string, options kapi.ListOptions) (runtime.Object, error) {
+			return &kapi.List{}, nil
+		},
+		ConstraintsFunc: imageStreamImportConstraintsFunc,
+	}
+}
+
+// imageStreamImportConstraintsFunc checks that given object is an image stream import.
+func imageStreamImportConstraintsFunc(required []kapi.ResourceName, object runtime.Object) error {
+	if _, ok := object.(*imageapi.ImageStreamImport); !ok {
+		return fmt.Errorf("unexpected input object %v", object)
+	}
+	return nil
+}
+
+// makeImageStreamImportAdmissionUsageFunc retuns a function for computing a usage of an image stream import.
+func makeImageStreamImportAdmissionUsageFunc(isNamespacer osclient.ImageStreamsNamespacer) generic.UsageFunc {
+	return func(object runtime.Object) kapi.ResourceList {
```

Comment:
Evaluators and ImageStream's strategy compute different resources. Evaluators evaluate `openshift.io/imagestreams` for resource quota controller/admission plugin. The strategy evaluates `openshift.io/{images,image-tags}`. So they don't interfere.

As for the IS evaluator and ISI evaluator - they don't interfere either because the ISI's REST uses internal image stream REST that bypasses admission chain.